src/components/screens/Post/components/PostHeader.tsx

```diff
@@ -34,6 +35,7 @@ interface IProps {
 
 function PostHeader({
   currentPost,
+  communityModerators,
```

Comment:
Since you do the same logic in two child components, from a parent component, just calculat ethe logic in the parent component. 


So you don't need `buildModList`. Replace that with `function isMod()`. It would take `userId` and `modList` (which would be like community.moderators). Then does the `some` function and returns a boolean. Then use that to calculate `isMod` to pass to `PostHeader` and `CommentItem`. Let me know if that makes sense.

Reply:
refactoring